x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -250,29 +259,52 @@ The Kibana Connector in use may need to be reconfigured with an updated Amazon B
     messages,
     model,
   }: InvokeAIActionParams): Promise<InvokeAIActionResponse> {
-    const res = await this.runApi({ body: JSON.stringify(formatBedrockBody({ messages })), model });
+    const res = await this.runApi({
+      body: JSON.stringify(formatBedrockBody({ messages, model })),
+      model,
+    });
     return { message: res.completion.trim() };
   }
 }
 
 const formatBedrockBody = ({
+  model = DEFAULT_BEDROCK_MODEL,
   messages,
+  stopSequences = ['\n\nHuman:'],
+  temperature = 0.5,
 }: {
+  model?: string;
   messages: Array<{ role: string; content: string }>;
+  stopSequences?: string[];
+  temperature?: number;
 }) => {
   const combinedMessages = messages.reduce((acc: string, message) => {
     const { role, content } = message;
-    // Bedrock only has Assistant and Human, so 'system' and 'user' will be converted to Human
-    const bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
+    const [, , modelName, majorVersion, minorVersion] =
+      (model || '').match(/(\w+)\.(.*)-v(\d+)(?::(\d+))?/) || [];
+    // Claude only has Assistant and Human, so 'user' will be converted to Human
+    let bedrockRole: string;
+
+    if (
+      role === 'system' &&
+      modelName === 'claude' &&
+      Number(majorVersion) >= 2 &&
+      Number(minorVersion) >= 1
+    ) {
+      bedrockRole = '';
+    } else {
+      bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
```

Comment:
I was curious what would happen if someone ended up using the old `DEFAULT_BEDROCK_MODEL` value as their model - I guess they would go through this path.  So seems safe to make a change to the default like this.